legacy_packages/sdk/src/evm/common/any-evm-utils/deployCreate2Factory.ts

```diff
@@ -21,44 +17,15 @@ export async function deployCreate2Factory(
   options?: DeployOptions,
 ): Promise<string> {
   invariant(signer.provider, "No provider");
-  const commonFactoryExists = await isContractDeployed(
-    COMMON_FACTORY,
+  const deploymentInfo = await computeCreate2FactoryTransaction(
     signer.provider,
   );
-  if (commonFactoryExists) {
-    return COMMON_FACTORY;
-  }
-
-  const enforceEip155 = await isEIP155Enforced(signer.provider);
-  const networkId = (await signer.provider.getNetwork()).chainId;
-  const chainId = enforceEip155 ? networkId : 0;
-  console.debug(`ChainId ${networkId} enforces EIP155: ${enforceEip155}`);
-  const deploymentInfo = CUSTOM_GAS_FOR_CHAIN[networkId]
-    ? getCreate2FactoryDeploymentInfo(chainId, {
-        gasPrice: CUSTOM_GAS_FOR_CHAIN[networkId].gasPrice,
-        gasLimit: CUSTOM_GAS_FOR_CHAIN[networkId].gasLimit,
-      })
-    : getCreate2FactoryDeploymentInfo(chainId, {});
-
-  const factoryExists = await isContractDeployed(
-    deploymentInfo.deployment,
-    signer.provider,
-  );
-
-  // deploy community factory if not already deployed
-  if (!factoryExists) {
-    const gasPrice = CUSTOM_GAS_FOR_CHAIN[networkId]?.gasPrice
-      ? CUSTOM_GAS_FOR_CHAIN[networkId].gasPrice
-      : 100 * 10 ** 9;
-    const gasLimit = CUSTOM_GAS_FOR_CHAIN[networkId]?.gasLimit
-      ? CUSTOM_GAS_FOR_CHAIN[networkId].gasLimit
-      : 100000;
-
-    invariant(gasLimit, "gasLimit undefined for create2 factory deploy");
-    invariant(gasPrice, "gasPrice undefined for create2 factory deploy");
 
+  if (deploymentInfo.transaction.length > 0) {
     // send balance to the keyless signer
-    const valueToSend = BigNumber.from(gasPrice).mul(gasLimit);
+    const valueToSend = BigNumber.from(deploymentInfo.valueToSend).mul(
+      100_000n,
```

Comment:
what is this 100k?

Reply:
Added comments.

Reply:
This is default deployment gas limit for create2 factory. 

https://github.com/Arachnid/deterministic-deployment-proxy?tab=readme-ov-file#deployment-gas-limit)